test/slice_matrices.jl

```diff
@@ -0,0 +1,134 @@
+using InteractiveUtils, SparseArrays, LinearAlgebra
+
+@testset "right_mul!" begin
+
+    # NOTE
+    # currently julia/sparsearrays does not implement a
+    # mul!(C::StridedMatrix, X::StridedMatrix, A::SparseMatrixCSC)
+    # instead it falls back to some generic matrix multiplciation in LinearAlgebra
+    #
+    # If this throws:
+    # - check if mul!() is now specialized for the above type signature
+    #   - if it is, remove right_mul! in helpers.jl & change right_muL! to mul! in
+    #     slice_matrices.jl
+    #   - otherwise, update the type signature
+    #
+    # see:
+    # Julia PR: https://github.com/JuliaLang/julia/pull/24045
+    # Julia Issue: https://github.com/JuliaLang/julia/issues/29956#issuecomment-440867024
+    # MC PR: https://github.com/crstnbr/MonteCarlo.jl/pull/9
+    # discourse: https://discourse.julialang.org/t/asymmetric-speed-of-in-place-sparse-dense-matrix-product/10256
+
+    M = rand(4, 4)
+    S = sparse(rand(4, 4))
+    X = similar(M)
+    mul_dense_sparse_is_missing = (@which mul!(X, M, S)).sig == Tuple{typeof(mul!),AbstractArray{T,2} where T,Union{AbstractArray{T,1}, AbstractArray{T,2}} where T,Union{AbstractArray{T,1}, AbstractArray{T,2}} where T}
```

Comment:
I like the idea of having a notification once the specialization lands in Julia. However, I wouldn't do it at the expense of inventing a new name `right_mul!` above. Also, the tests probably should fail because of this. The code works regardless of who defines the specialization (even if no ones defines it). Maybe it's better to throw a `@warn` in helpers.jl upon the definiton of the specialization in case it already exists in `LinearAlgebra`?

Reply:
Instead of creating fake `M, S, X` you can just call `which(mul!, (Matrix, Matrix, SparseMatrixCSC))`.

Reply:
Checking types, like you do, is one option. Another rather solid check could be
`occursin("SparseArrays", string(which(mul!, (Matrix, Matrix, SparseMatrixCSC)).file)) == false`

Reply:
I added such a warning now and switched to `mul!` and `occursin(...)`.